File: 2713-114-1769-minimum-number-of-operations-to-move-all-balls-to-each-box/2713-114-1769-minimum-number-of-operations-to-move-all-balls-to-each-box.py
class Solution:
    def minOperations(self, boxes: str) -> list[int]:
        n = len(boxes)
        op = 0
        L, R = 0, 0

        for i in range(n):
            if boxes[i] == '1':
                op += i
                R += 1

        v = [0] * n
        for i in range(n):
            v[i] = op
            if boxes[i] == '1':
                L += 1
                R -= 1
            op += -R + L

        return v

File: 2713-114-1769-minimum-number-of-operations-to-move-all-balls-to-each-box/2713-114-1769-minimum-number-of-operations-to-move-all-balls-to-each-box.cpp
class Solution {
public:
    vector<int> minOperations(string boxes) {
        int n = boxes.size(),op = 0 , L = 0 , R =0;
        for(int i = 0 ; i < n ;i++)
            if(boxes[i]=='1')op+=i,R++;
        vector<int>v(n);
        for(int i = 0 ; i < n ;i++){
            v[i] = op;
            if(boxes[i]=='1')L++,R--;
            op+=-R+L;
        }
        return v;
    }
};